### PR TITLE
test: Add test validating that BLOCKHASH returns the latest 256 blocks

### LINKED ISSUE #3336: BLOCKHASH returns the hash of the latest 256 blocks
- Write an E2E test for `BLOCKHASH` returns the hash of the latest 256 blocks
- From: https://www.notion.so/swirldslabs/c306bbf842264eb391a9ee4cc92e1639?v=94195c43eba64dc2a39d79fc93c838de

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:

**Related issue(s)**:

Fixes #3336 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by MrValioBg — APPROVED** [review]
Thank you very much @stoyanov-st !

LGTM!

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java [hunk 1]
```diff
@@ -299,6 +311,55 @@ final Stream<DynamicTest> blck001And002And003And004ReturnsCorrectBlockProperties
...
+                    assertTrue(blockHashRes
+                            .getResponseRecord()
+                            .getContractCallResult()
+                            .getErrorMessage()
+                            .isEmpty());
+                    final var res = blockHashRes
+                            .getResponseRecord()
+                            .getContractCallResult()
+                            .getContractCallResult()
+                            .substring(64);
```
Comment (david-bakin-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java:343]: What's in the first 8 bytes you're lopping off here? (I think that's what's going on.)

**Review by david-bakin-sl — APPROVED** [review]
Looks good.  SOme questions, and also a suggestion about renaming the suite, if you'd like to.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java (hunk 1 above)
Comment (stoyanov-st) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java:343]: I trim the headers so I can have only the result bytes

Comment (david-bakin-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java:343]: oh! what's in the headers? (I didn't know there _were_ headers ...)

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java [hunk 2]
```diff
@@ -59,13 +68,16 @@
...
+import java.util.function.Consumer;
 import java.util.stream.Stream;
 import org.apache.tuweni.bytes.Bytes32;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.DynamicTest;
 import org.junit.jupiter.api.Tag;
 
 @Tag(SMART_CONTRACT)
+@DisplayName("Records Suite")
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java:80]: I'm not sure about the naming suggestion - the suite is in contract.records package and validates the records from contract/ethereum calls. If you could suggest a more descriptive one?

About the `bigCall` test, it just returns a bytes array with the length of the input, no clue why, though?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM!

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-bakin-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java:80]: OK, NVM about the name change.  You're right the package name suffices to clarify the class name.

About `bigCall` - I was going to ask the "author" what was going on here but how is it possible that `git blame` shows 8 separate authors for a 15 line test method that does almost nothing?  And none of those authors thought to put in a comment. (Including, unfortunately: me.)

**Pull request merged**